### PR TITLE
Created the SBT contract

### PR DESCRIPTION
**Pull request by gaurangtorvekar**
Responding to OpenOasis issue https://github.com/Cypher-Laboratory/OpenOasis/issues/2

This contract implements an SBT with all the required functionality but uses a dummy signature verification contract because such a contract is not available yet.

### CODE REVIEW

File: EVM-verifier/contracts/AliceRingToken.sol [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        MINTED // the proof has already been minted
+    }
+
+    error AlreadyMinted(string proofId);
+    error InvalidSignature();
+    error InvalidTokenAmounts();
+    error OnlyOwnerCanBurn(uint256 tokenId);
+
+    RingSignatureVerifier public verifier; // Instance of the ring signature verifier contract
+    uint256 public minimalAmount; // Minimal amount of tokens to be owned by each address in the ring
```
Comment (Elli610) [inline EVM-verifier/contracts/AliceRingToken.sol]: This should not be a global variable. Each SBT proof can have a different token value. 
`minimalAmount` should be a function variable

File: EVM-verifier/contracts/AliceRingToken.sol [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    mapping(string => Status) public mintStatus; // signatureHash => Status (computed off-chain)
+
+    constructor(address initialOwner, address _verifier, uint256 _minimalAmount)
+        ERC721("AliceRingToken", "ART")
+        Ownable(initialOwner)
```
Comment (Elli610) [inline EVM-verifier/contracts/AliceRingToken.sol]: `Ownable`  does not take any argument

**Review by Elli610 — COMMENTED** [review]
- I cannot compile the contract with solidity 0.8.20 because `_burn`  should be implemented
- Please add some function documentation on top of each. (using the same template we used in our [contract architecture definition](https://github.com/orgs/Cypher-Laboratory/discussions/2))

Otherwise, LGTM! 
@LeJamon do you see anything else ?

File: EVM-verifier/contracts/AliceRingToken.sol (hunk 1 above)
Comment (Elli610) [reply EVM-verifier/contracts/AliceRingToken.sol]: it should be an argument of `mint`. Our bad, we forgot it in the contract architecture

**Review by Elli610 — COMMENTED** [review]
closes #2

**Review by Elli610 — CHANGES_REQUESTED** [review]
changes needed

File: EVM-verifier/contracts/AliceRingToken.sol (hunk 2 above)
Comment (gaurangtorvekar) [reply EVM-verifier/contracts/AliceRingToken.sol:49]: It does in the latest v5 implementation of Openzeppelin. If you use an earlier version, then there's no argument.
Check this out - https://blog.openzeppelin.com/introducing-openzeppelin-contracts-5.0

**Comment by gaurangtorvekar** [comment]
@Elli610 all your comments have been resolved now. Please check...

File: EVM-verifier/contracts/AliceRingToken.sol [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+    /**
+     * @notice Verifies the token amounts for each address in the ring
+     *
+     * @param token - the erc20 address of the token we are proving the ownership of
+     * @param signature - the signature object
+     */
+    function verifyTokenAmounts(address token, uint256 minBalance, Signature memory signature)
+        internal
+        returns (bool)
```
Comment (Elli610) [inline EVM-verifier/contracts/AliceRingToken.sol]: - `@param minBalance` is missing
- this function can be restricted to view

**Review by Elli610 — CHANGES_REQUESTED** [review]
once `verifyTokenAmounts` is fixed, we should be all good !

**Review by Elli610 — COMMENTED** [review]
@LeJamon Is it ok for you ?

File: EVM-verifier/contracts/AliceRingToken.sol (hunk 3 above)
Comment (gaurangtorvekar) [reply EVM-verifier/contracts/AliceRingToken.sol]: @Elli610 done! Silly that I missed this, thanks for pointing out...

**Review by Elli610 — APPROVED** [review]

**Pull request merged**